packages/ember-data/lib/system/model/model.js

```diff
@@ -473,12 +473,31 @@ var Model = Ember.Object.extend(Ember.Evented, {
   },
 
   _setup: function() {
+    var record = this;
     this._changesToSync = {};
     this._deferredTriggers = [];
     this._data = {};
     this._attributes = Ember.create(null);
     this._inFlightAttributes = Ember.create(null);
-    this._relationships = {};
+    this._relationships = Ember.create({
+      initializedRelationships: {},
```

Comment:
Instead of `{}` this should be `Ember.create(null)` to prevent clashes in the unlikely event that a relationship shares a name with a property on the Object prototype. You may also want to move this into an init function of you make a `Relationships` object like @fivetanley suggests above.